workspaces/acs/plugins/src/components/DataFilterComponent/DataFilterComponent.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import { EntitySelectComponent } from './EntitySelectComponent';
+import { AttributeSelectComponent } from './AttributeSelectComponent';
+import { InputFieldComponent } from './InputFieldComponent';
+import { CheckboxSelectComponent } from './CheckboxSelectComponent';
+import {
+  Toolbar,
+  ToolbarGroup,
+  ToolbarItem,
+  ToolbarContent,
+} from '@patternfly/react-core';
+import { useTheme } from '@material-ui/core/styles';
+
+export const DataFilterComponent = ({ setFilters, data }) => {
+  const attributes = ['Name', 'Discovered time', 'CVSS'];
+  const entities = {
+    Image: ['Name'],
+    CVE: ['Name', 'Discovered time', 'CVSS'],
+    'Image Component': ['Name'],
+    Deployment: ['Name'],
+    Namespace: ['Name'],
+    Cluster: ['Name'],
+  };
+
+  const cveSeverityOptions = ['Critical', 'Important', 'Moderate', 'Low'];
+  const cveStatusOptions = ['Fixable', 'Not fixable'];
+
+  const theme = useTheme();
+  const isDarkMode = theme.palette.type === 'dark';
+
+  const [selectedEntity, setSelectedEntity] = useState('Image');
+  const [selectedAttribute, setSelectedAttribute] = useState('Name');
+
+  const [userText, setUserText] = useState('');
+  const [selectedCveSeverityOptions, setSelectedCveSeverityOptions] = useState(
+    [],
+  );
+  const [selectedCveStatusOptions, setSelectedCveStatusOptions] = useState([]);
+
+  const getSelectedAttributes = () => {
+    return entities[selectedEntity];
+  };
+
+  const modifyPFCardStyle = () => {
+    const style = document.createElement('style');
+    style.id = 'filter-group-styles';
+    style.innerHTML = `
```

Comment:
Is there any way we can avoid this? This will cause issues with custom themes having a hard coded color